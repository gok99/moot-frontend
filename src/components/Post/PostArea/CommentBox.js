import React, { useState } from 'react';
import { Row, Col, Button, Form } from 'react-bootstrap';

import { convertTime } from '../../utils.js';

import '../../Styles/styles.css';
import '../post.css';

const CommentBox = (props) => {
  const fb = props.fb;
  const uid = props.uid;
  const postUid = props.postUid;
  const poster = props.poster;
  const commentedPostKey = props.commentedPostKey;
  const [editState, setEditState] = useState(false);

  const commentKey = props.commentKey;
  const commentContent = props.commentContent;
  const [currentComment, setCurrentComment] = useState(commentContent);
  const commentTime = props.commentTime;
  const myComment = props.myComment;
  const newTime = convertTime(commentTime);

  const deleteComment = (event) => {
    fb.deletePostUserComments(postUid, commentKey);
    fb.deleteUserCommentedPosts(uid, commentedPostKey);
    event.preventDefault();
  };

  const editComment = (event) => {
    setEditState(true);
    event.preventDefault();
  };

  const onSubmit = (event) => {
    fb.postUserComments(postUid).child(commentKey).update({
      comment: currentComment
    })
    fb.userCommentedPosts(uid).child(commentedPostKey).update({
      comment: currentComment
    })
    setEditState(false);
    event.preventDefault();
  }

  const onChange = (event) => {
    setCurrentComment(event.target.value);
  };

  return (
    <>
      <Row>
        <Col md={8} className="d-flex justify-content-start">
          <p className="text-post meta">Comment by {poster} on {newTime}:</p>
        </Col>
        <Col md={4} className="d-flex justify-content-end">
        {
          myComment
            ? <>
                <Button className="btn-postexpand comment" type="button" onClick={editComment}>
                  Edit
                </Button>
                <Button className="btn-postexpand comment" type="button" onClick={deleteComment}>
                  Delete
                </Button>
              </>
            : null
        }
        </Col>
      </Row>
      { editState
          ? <Row className="mb-3">
              <Form className="mt-2" onSubmit={onSubmit}>
                  <Row>
                    <Col>
                      <Form.Group controlId="comment">
                        <Form.Control
                          className="input-comment"
                          name="currentComment" 
                          type="text"
                          as="textarea"
                          placeholder="Edit your comment!"
                          value={currentComment}
                          onChange={onChange} />
                      </Form.Group>
                    </Col>
                    <Col md="auto">
                      <Button  
                        className="btn-postcreation btn-comment"
                        type="submit"
                        disabled={currentComment === ''}>
                        Edit Comment
                      </Button>
                    </Col>
                  </Row>
                </Form>
              </Row>
          : <Row className="b-text-commentcontent mb-3"><p className="text-post content">{commentContent}</p></Row>
      }
    </>
  );
};

export default CommentBox;