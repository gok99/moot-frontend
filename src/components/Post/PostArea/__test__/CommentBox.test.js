import React from 'react';
import ReactDOM from 'react-dom';
import CommentBox from '../CommentBox';

import { cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<CommentBox  />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<CommentBox  />).toJSON();
  expect(tree).toMatchSnapshot();
});