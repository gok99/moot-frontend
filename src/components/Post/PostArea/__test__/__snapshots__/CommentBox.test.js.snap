// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="d-flex justify-content-start col-md-8"
    >
      <p
        className="text-post meta"
      >
        Comment by 
         on 
        NaN Invalid Date, NaN:NaN AM
        :
      </p>
    </div>
    <div
      className="d-flex justify-content-end col-md-4"
    />
  </div>,
  <div
    className="mb-3 row"
  >
    <p
      className="text-post content"
    />
  </div>,
]
`;
